Attached clusters data sources

hankfreund opened the pull request:
* Added new datasource for retrieving valid attached cluster platform versions `google_container_attached_versions`
* Added new datasource for retrieving a manifest to apply to the user cluster for boot-strapping `google_container_attached_install_manifest`

<!--
Replace each [ ] with [X] to check it. Switch to the preview view to make it easier to click on links.
These steps will speed up the review process, and we appreciate you spending time on them before sending
your code to be reviewed.
-->
If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

<!-- AUTOCHANGELOG for Downstream PRs.

Please select one of the following "release-note:" headings:
    - release-note:enhancement
    - release-note:bug
    - release-note:note
    - release-note:new-resource
    - release-note:new-datasource
    - release-note:deprecation
    - release-note:breaking-change
    - release-note:none
    
Unless you choose release-note:none, please add a release note.

See .ci/RELEASE_NOTES_GUIDE.md for writing good release notes.

You can add more release note blocks if you want more than one CHANGELOG
entry for this PR.
-->
**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-datasource
google_container_attached_versions
```

```release-note:new-datasource
google_container_attached_install_manifest
```

modular-magician commented:
Hello!  I am a robot who works on Magic Modules PRs.

I've detected that you're a community contributor. @scottsuarez, a repository maintainer, has been assigned to assist you and help review your changes. 

<details>
  <summary>:question: First time contributing? Click here for more details</summary>

---

Your assigned reviewer will help review your code by: 
* Ensuring it's backwards compatible, covers common error cases, etc.
* Summarizing the change into a user-facing changelog note.
* Passes tests, either our "VCR" suite, a set of presubmit tests, or with manual test runs.

You can help make sure that review is quick by running local tests and ensuring they're passing in between each push you make to your PR's branch. Also, try to leave a comment with each push you make, as pushes generally don't generate emails.

If your reviewer doesn't get back to you within a week after your most recent change, please feel free to leave a comment on the issue asking them to take a look! In the absence of a dedicated review dashboard most maintainers manage their pending reviews through email, and those will sometimes get lost in their inbox.

---

</details>

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7086-old..auto-pr-7086) ( 8 files changed, 321 insertions(+), 7 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7086-old..auto-pr-7086) ( 8 files changed, 321 insertions(+), 7 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-7086-old..auto-pr-7086) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-7086-old..auto-pr-7086) ( 2 files changed, 12 insertions(+), 2 deletions(-))

modular-magician commented:
#### Tests analytics 
Total tests: `2393` 
Passed tests `2135` 
Skipped tests: `248` 
Failed tests: `10` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccFirebaserulesRelease_BasicRelease|TestAccDatastreamStream_datastreamStreamFullExample|TestAccDatastreamStream_update|TestAccDatastreamStream_datastreamStreamBasicExample|TestAccContainerCluster_withInvalidGatewayApiConfigChannel|TestAccContainerAttachedCluster_update|TestAccContainerAttachedCluster_containerAttachedClusterFullExample|TestAccContainerAttachedCluster_containerAttachedClusterBasicExample|TestAccComputeForwardingRule_update|TestAccDataSourceGoogleContainerAttachedInstallManifest </blockquote></details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/753cbf0d-d80d-404a-bc87-d94bb9af666d/recording/TestAccFirebaserulesRelease_BasicRelease.log)]
`TestAccDatastreamStream_datastreamStreamFullExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/753cbf0d-d80d-404a-bc87-d94bb9af666d/recording/TestAccDatastreamStream_datastreamStreamFullExample.log)]
`TestAccDatastreamStream_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/753cbf0d-d80d-404a-bc87-d94bb9af666d/recording/TestAccDatastreamStream_update.log)]
`TestAccDatastreamStream_datastreamStreamBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/753cbf0d-d80d-404a-bc87-d94bb9af666d/recording/TestAccDatastreamStream_datastreamStreamBasicExample.log)]
`TestAccContainerCluster_withInvalidGatewayApiConfigChannel`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/753cbf0d-d80d-404a-bc87-d94bb9af666d/recording/TestAccContainerCluster_withInvalidGatewayApiConfigChannel.log)]
`TestAccContainerAttachedCluster_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/753cbf0d-d80d-404a-bc87-d94bb9af666d/recording/TestAccContainerAttachedCluster_update.log)]
`TestAccContainerAttachedCluster_containerAttachedClusterFullExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/753cbf0d-d80d-404a-bc87-d94bb9af666d/recording/TestAccContainerAttachedCluster_containerAttachedClusterFullExample.log)]
`TestAccContainerAttachedCluster_containerAttachedClusterBasicExample`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/753cbf0d-d80d-404a-bc87-d94bb9af666d/recording/TestAccContainerAttachedCluster_containerAttachedClusterBasicExample.log)]
`TestAccComputeForwardingRule_update`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/753cbf0d-d80d-404a-bc87-d94bb9af666d/recording/TestAccComputeForwardingRule_update.log)]
`TestAccDataSourceGoogleContainerAttachedInstallManifest`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/753cbf0d-d80d-404a-bc87-d94bb9af666d/recording/TestAccDataSourceGoogleContainerAttachedInstallManifest.log)] 

All tests passed
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/753cbf0d-d80d-404a-bc87-d94bb9af666d/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/753cbf0d-d80d-404a-bc87-d94bb9af666d/recording) for each test

ScottSuarez commented:
Hey ! wanted to check in. Is there a corresponding issues for these datasources being added? Looking for more information on the need and reason for why. internal or external

hankfreund commented:
Hi! Sorry for the confusion. These datasources round out the required implementation for the new attached clusters API. You can see my previous PR for adding the resource here: https://github.com/GoogleCloudPlatform/magic-modules/pull/6858
There is not a corresponding external issue for this. We can chat offline if there's any other info you need.

ScottSuarez commented on mmv1/third_party/terraform/data_sources/data_source_google_container_attached_install_manifest.go:
```diff
@@ -0,0 +1,71 @@
+)
+
+func dataSourceGoogleContainerAttachedInstallManifest() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceGoogleContainerAttachedInstallManifestRead,
+		Schema: map[string]*schema.Schema{
+			"location": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"parent": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cluster_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"platform_version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"manifest": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
```
Switching from "parent" to `project`, `region` might be a better UX. If region is identical to location this `region` can be omitted.

ScottSuarez commented on mmv1/templates/terraform/examples/container_attached_cluster_basic.tf.erb:
```diff
@@ -1,6 +1,11 @@
 data "google_project" "project" {
```
Both data sources could use a datasource markdown file.

Checkout https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/website/docs/d

ScottSuarez commented on mmv1/templates/terraform/examples/container_attached_cluster_basic.tf.erb:
```diff
@@ -1,6 +1,11 @@
 data "google_project" "project" {
 }
```
I would prefer splitting this down to two PRs.. one for each data source. To keep the context more focused. It doesn't seem like the data sources are dependent on eachother.

ScottSuarez reviewed (changes requested):
Seems fairly straightforward.. Here are some initial thoughts.

modular-magician commented:
Hi there, I'm the Modular magician. I've detected the following information about your changes:

## Diff report 
Your PR generated some diffs in downstreams - here they are.

Terraform GA: [Diff](https://github.com/modular-magician/terraform-provider-google/compare/auto-pr-7086-old..auto-pr-7086) ( 7 files changed, 225 insertions(+), 7 deletions(-))
Terraform Beta: [Diff](https://github.com/modular-magician/terraform-provider-google-beta/compare/auto-pr-7086-old..auto-pr-7086) ( 7 files changed, 225 insertions(+), 7 deletions(-))
TF Validator: [Diff](https://github.com/modular-magician/terraform-validator/compare/auto-pr-7086-old..auto-pr-7086) ( 2 files changed, 3 insertions(+), 3 deletions(-))
TF OiCS: [Diff](https://github.com/modular-magician/docs-examples/compare/auto-pr-7086-old..auto-pr-7086) ( 2 files changed, 12 insertions(+), 2 deletions(-))

hankfreund commented:
I've split the PR into two. This one is now only for `google_container_attached_versions` and test modifications to use it.
Addressed the review comments.

modular-magician commented:
#### Tests analytics 
Total tests: `2400` 
Passed tests `2143` 
Skipped tests: `251` 
Failed tests: `6` 

#### Action taken 
<details> <summary>Triggering VCR tests in RECORDING mode for the tests that failed during VCR. Click here to see the failed tests</summary><blockquote>TestAccFirebaserulesRelease_BasicRelease|TestAccContainerCluster_withInvalidGatewayApiConfigChannel|TestAccDataSourceSqlDatabaseInstances_tierFilter|TestAccDataSourceSqlDatabaseInstances_regionFilter|TestAccDataSourceSqlDatabaseInstances_databaseVersionFilter|TestAccDataSourceSqlDatabaseInstances_basic </blockquote></details>

modular-magician commented:
Tests passed during RECORDING mode:
 `TestAccFirebaserulesRelease_BasicRelease`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/b496bd20-47eb-49af-bf4f-61875f5fb40c/recording/TestAccFirebaserulesRelease_BasicRelease.log)]
`TestAccContainerCluster_withInvalidGatewayApiConfigChannel`[[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/b496bd20-47eb-49af-bf4f-61875f5fb40c/recording/TestAccContainerCluster_withInvalidGatewayApiConfigChannel.log)] 

Tests failed during RECORDING mode:
 `TestAccDataSourceSqlDatabaseInstances_tierFilter`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/b496bd20-47eb-49af-bf4f-61875f5fb40c/build-log/recording_build/TestAccDataSourceSqlDatabaseInstances_tierFilter_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/b496bd20-47eb-49af-bf4f-61875f5fb40c/recording/TestAccDataSourceSqlDatabaseInstances_tierFilter.log)]
`TestAccDataSourceSqlDatabaseInstances_regionFilter`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/b496bd20-47eb-49af-bf4f-61875f5fb40c/build-log/recording_build/TestAccDataSourceSqlDatabaseInstances_regionFilter_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/b496bd20-47eb-49af-bf4f-61875f5fb40c/recording/TestAccDataSourceSqlDatabaseInstances_regionFilter.log)]
`TestAccDataSourceSqlDatabaseInstances_databaseVersionFilter`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/b496bd20-47eb-49af-bf4f-61875f5fb40c/build-log/recording_build/TestAccDataSourceSqlDatabaseInstances_databaseVersionFilter_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/b496bd20-47eb-49af-bf4f-61875f5fb40c/recording/TestAccDataSourceSqlDatabaseInstances_databaseVersionFilter.log)]
`TestAccDataSourceSqlDatabaseInstances_basic`[[Error message](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/b496bd20-47eb-49af-bf4f-61875f5fb40c/build-log/recording_build/TestAccDataSourceSqlDatabaseInstances_basic_recording_test.log)] [[Debug log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/b496bd20-47eb-49af-bf4f-61875f5fb40c/recording/TestAccDataSourceSqlDatabaseInstances_basic.log)] 

Please fix these to complete your PR
View the [build log](https://storage.cloud.google.com/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/b496bd20-47eb-49af-bf4f-61875f5fb40c/build-log/recording_test.log) or the [debug log](https://console.cloud.google.com/storage/browser/ci-vcr-logs/beta/refs/heads/auto-pr-7086/artifacts/b496bd20-47eb-49af-bf4f-61875f5fb40c/recording) for each test
